### DISCUSSION TITLE
Action names must match function name -> source of error

### DISCUSSION THREAD

**Discussion by DavideWalder** [zitadel/zitadel#4783] [Ideas]
This adds complexity since they must stay in sync (typos, bad copy-paste, ...)

This could be simplified as k6 does. That is, the action can be the default exported function. This ensures that it is unique and it doesn't require the use of a specific name.

```js
export function someHelper() {
  // ...
}

// the action
export default function () {
  someHelper();
}
```
See https://k6.io/docs/using-k6/modules/

**Comment by adlerhurst** [comment]
I like the idea and it would definitely improve the usability of actions 👍

**Reply by fforootd** [reply]
I agree this is a must for our v2 discussion #4265

**Reply by fforootd** [reply]
@hifabienne i think we should fix this 

It leads to a problem if a user deploys an action that cant be called, this can break the login without a possibility of getting to login into zitadel.
